### ISSUE TITLE
Cannot install some C-packages via standard Python tooling

### ISSUE THREAD

**Issue by dstufft** [ContinuumIO/anaconda-issues#177]
(Ana)conda breaks the ability to do things like `pip install cryptography`. It does this I believe because it does not set the `LD_LIBRARY_PATH` on Linux or the `DYLD_LIBRARY_PATH` path on OS X.

While this is exposed with cryptography, it isn't unique to it. It was originally reported as https://github.com/conda/conda-recipes/issues/110 and on that ticket people also pointed out that it affected psycopg2.

The solution on https://github.com/conda/conda-recipes/issues/110 was to package cryptography and psycopg2 for (Ana)conda however I do not believe that is a satisfactory solution. The problem will still arise anytime someone needs to install a different package that runs into this same problem _and_ it introduces a serious regression in the ability for people to rely on projects like cryptography and psycopg2.

If I am the author of a project that depends on cryptography, I can have one command that works cross platform to install my project which is `pip install thing-that-depends-on-cryptography`. The user will have to ensure that they have OpenSSL installed (and a compiler on OSX/Linux), but that's it. However if their Python comes from (Ana)conda, then they'll also need to set some environment variables or pass some flags in order to instruct pip where to locate things from.

The end result is that the cryptography project (and any other project that suffers from the same thing) and  _all_ of the projects that depend on it, no matter how deep in the dependency graph, needs to add explicit instructions for Conda saying "oh and if you want to install this on Conda, first do `conda install cryptography`" _or_ they need to package Conda packages which many of those projects won't do, just like they don't release RPMs or DEBs or what have you.

Just to be clear, I'm fairly sure this isn't specific to pip, the same problem will come from easy_install/setuptools, `setup.py install`, and distutils since pip essentially just dispatches to those things under the cover and the error is realling bubbling up from those systems.

**Comment by dan-blanchard** [comment]
I just ran into this issue a couple days ago.  Just to summarize, the workaround is that we set both `CFLAGS` and `DYLD_LIBRARY_PATH` when running `pip install` and everything should be good after that, right?

**Comment by asmeurer** [comment]
Once https://github.com/conda/conda-build/pull/312 is done and we rebuild Python and openssl with it, this issue should go away.

**Comment by asmeurer** [comment]
The best fix here is to use a conda recipe (if it's specifically cryptography, there already is a conda package for it). conda build will do things that negate this issue. Setting `DYLD_LIBRARY_PATH` (or better, `DYLD_FALLBACK_LIBRARY_PATH` will work, but you may have to keep it set forever to use the package). If you can get the right CFLAGS set, it may work thereafter (I haven't tested this scenario very much).

**Comment by Juanlu001** [comment]
Is this issue still present?

**Comment by kalefranz** [comment]
I think this issue has long been put to rest with conda/conda-build#312.
